engine/src/main/scala/cromwell/webservice/WorkflowQueryPagination.scala

```diff
@@ -0,0 +1,41 @@
+package cromwell.webservice
+
+import cromwell.services.MetadataServiceActor.QueryMetadata
+import spray.http.HttpHeaders.Link
+import spray.http.{HttpHeader, Uri}
+
+
+trait WorkflowQueryPagination {
+
+  protected def generatePaginationParams(page: Int, pageSize: Int): String = {
+    s"page=$page&pagesize=$pageSize"
+  }
+
+  //Generates link headers for pagination navigation https://tools.ietf.org/html/rfc5988#page-6
+  protected def generateLinkHeaders(uri: Uri, metadata: Option[QueryMetadata]): Seq[HttpHeader] = {
+    //strip off the query params
+    val baseUrl = uri.scheme + ":" + uri.authority + uri.path
+    metadata match {
+      case Some(meta) =>
+        (meta.page, meta.pageSize) match {
+          case (Some(p), Some(ps)) =>
+
+            val firstLink = Link(Uri(baseUrl).withQuery(generatePaginationParams(1, ps)), Link.first)
```

Comment:
No tests seem to be checking this, so I'm not sure.. but... if one uses a URL like in the README, what is the value of headers `firstLink`, etc. after one uses a URL like these?

- `curl "http://localhost:8000/api/workflows/v1/query?start=2015-11-01&end=2015-11-03&status=Failed&status=Succeeded&page=1&pagesize=10"`<sup>1</sup>

- `curl -X POST --header "Content-Type: application/json" -d "[{\"start\": \"2015-11-01\"}, {\"end\": \"2015-11-03\"}, {\"status\": \"Failed\"}, {\"status\": \"Succeeded\"}, {\"page\": "1"}, {\"pagesize\": "10"}]" "http://localhost:8000/api/workflows/v1/query"`

Looking at the code flow, for GET it seems to probably drop parameters like `start` and `status`. For POST, it seems to add parameters that were supposed to be in the POST content, not the query parameters.

<sup>1</sup> There's an extra " in the README after the cURL and HTTPie examples for GET /api/workflows/:version/query.